src/bot.py

```diff
@@ -222,18 +226,19 @@ def on_init(self):
     def on_select_character(self):
         if self._config.general['restart_d2r_when_stuck']:
             # Make sure the correct char is selected
-            if self._char_selector.has_char_template_saved():
-                self._char_selector.select_char()
+            if SelectedCharacter.has_char_template_saved():
+                SelectCharacter(self._template_finder).select_char()
             else:
```

Comment:
```suggestion
            selector = SelectCharacter(self._template_finder)
            if selector.has_char_template_saved():
                selector.select_char()
            else:
```

Reply:
I understand that has_char_templated_saved is class level but in this context it's nicer to use it as instance based but still have your storage at class level

Reply:
got it, changed, thanks